File: leetcode/merge-strings-alternately.js
function mergeAlternately(word1, word2) {
    let ret = "";
    const maxLength = Math.max(word1.length, word2.length);
    
    for (let i = 0; i < maxLength; i++) {
        if (i < word1.length) ret += word1[i];
        if (i < word2.length) ret += word2[i];
    }
    
    return ret;
}

File: leetcode/merge-strings-alternately.dart
String mergeAlternately(String word1, word2) {
  String ret = "";

  for (int i = 0; i < word1.length; i++) {
    ret += "${word1[i]}${word2[i]}";
  }

  if (word2.length - word1.length > 0) {
    ret += word2.substring(word2.length - word1.length);
  }
  

  return ret;
}

void main() {}
